spring-rabbit/src/main/java/org/springframework/amqp/rabbit/annotation/RabbitListenerAnnotationBeanPostProcessor.java

```diff
@@ -265,6 +266,15 @@ private Method checkProxy(Method method, Object bean) {
 				// Found a @RabbitListener method on the target class for this JDK proxy ->
 				// is it also present on the proxy itself?
 				method = bean.getClass().getMethod(method.getName(), method.getParameterTypes());
+				Class<?>[] proxiedInterfaces = ((Advised) bean).getProxiedInterfaces();
+				for (Class<?> iface : proxiedInterfaces) {
+					try {
+						method = iface.getMethod(method.getName(), method.getParameterTypes());
+						break;
+					}
+					catch (NoSuchMethodException noMethod) {
+					}
```

Comment:
Well, if there is no other way to do the same (and I guess no :smile: ), I don't see reason to wait for @snicoll .
Although it looks like need to debug the stuff. Because you have changed the previous method for the `MethodRabbitListenerEndpoint` and thereafter for the `processListener()`...